test/mocha/no-mutate-input.js

```diff
@@ -0,0 +1,33 @@
+import assert from "assert";
+import { minify } from "../../main.js";
+
+describe("no-mutate-input", function() {
+
+    it("does not modify the options object", async function() {
+        const originalConfig = {
+            format: {},
+            sourceMap: true,
+        };
+        const config = {...originalConfig};
```

Comment:
```suggestion
        const config = {
            format: {},
            sourceMap: true,
        };
```

Using a shallow copy might hide the failure (config.format being mutated).